.github/workflows/release.yml

```diff
@@ -55,6 +55,50 @@ jobs:
           path: dist/
       - uses: pypa/gh-action-pypi-publish@15c56dba361d8335944d31a2ecd17d700fc7bcbc # v1.12.2
 
+  build_and_push_container:
+    name: Build and Push Docker Image
+    needs: [publish_release_to_pypi]
+    runs-on: ubuntu-latest
+    permissions:
+      id-token: write
```

Comment:
since we include this permission, we should sign the container image with `actions/attest-build-provenance`
which requires an additional permission: `attestations: write`